### ISSUE TITLE
Hotspot continuing to lose sync

### ISSUE THREAD

**Issue by clemsonteg** [helium/miner#1440]
One of my RAK V2 hotspots is unable to maintain sync. 

The network and connection are all good and have been checked directly at the ethernet cable connected to the hotspot. Diagnostics via bluetooth show everything to be working and the hotspot is gaining in block height, but at a rate slower than the blockchain. 

I have two other hotspots in the area that use the same ISP and router at each location. They are continuing to participate in network activity, so I don't believe it is internet related. 

The SD card was replaced approximately one month ago with a Sandisk 64gb High Endurance card, following the instructions from RAK/MNTD. The hotspot has operated flawlessly for the last few weeks.

In comparing the block height reported in diagnostics over the last 24 hours to the actual chain height at the time of the diagnostics report, the hotspot is losing sync status rather than gaining. I've attached an image of the data collected. 

![gap](https://user-images.githubusercontent.com/99567495/154302150-499ce894-3323-4663-acc7-410dbf659f35.png)

Peers reported via Hikerbot peerbook all seem to be in good standing, connected to the network, and continuing to participate in activity.

I have not power cycled the hotspot at this point. 

Looking for advice on how to move forward and to see if others are experiencing this.

**EDIT**
As a follow-up, my hotspot was not able to close the sync gap, even after the release of an updated snapshot. It remained at approximately 230 blocks behind for several hours. I re-formatted another card I had lying around, flashed it following MNTD's instructions, and added it to the hotspot. Within 4 hours of doing so, the hotspot was fully synced and created a challenge. 

Even though the card that was in the hotspot was the recommended size, brand, and model, I think something about the process of clearing data and/or resetting to the last snapshot was not working properly. I don't have anything concrete to support that other than replacing the card worked without issue.

**Comment by serbyxp** [comment]
Yeah that happened to me last night, I re installed and deleted the blockchain.db folder, it got stuck in a loop that it wouldn’t fix automatically, it kept throwing a bunch of errors, something about the blockchain expected Xnumber and calculated way Ynumber, some other errors to, I’m not at the machine right now I will post those errors when I get back

**Comment by absolutebica** [comment]
I am in the same boat.  

Rak blackspot, running the latest firmware (I flashed 01.04 from mntd, it auto updated to 02.10) and my gap continues to widen.  After the snapshot push yesterday I went from 1900 behind to around 480. Then I flashed the 01.04 firmware because I had previously used the helium firmware back in mid January (though the miner performed just fine up until Feb 13).  My sync gap has risen from 500 to 900 by this morning (feb 17, 9:45am).  

I have a ticket open on Rakstars but so far it hasn't really gotten looked at.  

I personally think this is a bigger widespread issue and not enough people are in these groups to post their issue, but that is just an assumption.  

But something is going on with Rak's that is causing them to stall and not update with the blockchain.   

I have 2 blackspot raks and they both have been dead flat since Feb 13.

**Comment by mojomartini** [comment]
> I personally think this is a bigger widespread issue and not enough people are in these groups to post their issue, but that is just an assumption.

Hey,  @absolutebica ,
Former (c**c***.com) co-worker, Dan S. here.
I also believe it's more widespread than reported but reporting is another issue.

Here's one thing to look at if it applies in your situation.
At one point I was using PureVPN so I could port forward with a public IP.  In testing, despite what seemed to be a decent connection, whenever I was connected to the VPN, it would start lagging on blockchain updates and the gap would start widening.  Whenever I disconnected the VPN, the gaps decreased. I'm pretty sure it wasn't a bandwidth issue.  I would also tail the logs when doing both tests to see any difference but did not.  
In short, it seems like low-quality VPNs might be contributing to some of these issues.

Again, this won't fit everyone's scenario but it's something to consider.

**Comment by absolutebica** [comment]
> > I personally think this is a bigger widespread issue and not enough people are in these groups to post their issue, but that is just an assumption.
> 
> Hey, @absolutebica , Former (c**c***.com) co-worker, Dan S. here. I also believe it's more widespread than reported but reporting is another issue.
> 
> Here's one thing to look at if it applies in your situation. At one point I was using PureVPN so I could port forward with a public IP. In testing, despite what seemed to be a decent connection, whenever I was connected to the VPN, it would start lagging on blockchain updates and the gap would start widening. Whenever I disconnected the VPN, the gaps decreased. I'm pretty sure it wasn't a bandwidth issue. I would also tail the logs when doing both tests to see any difference but did not. In short, it seems like low-quality VPNs might be contributing to some of these issues.
> 
> Again, this won't fit everyone's scenario but it's something to consider.

I'm not on a VPN.  In fact, I'm on gigabit internet, currently doing about 800Mbps down, 800Mbps up.

**Comment by yuseaname** [comment]
My hot spot went offline yesterday (Rak mntd gold spot) not on VPN, haven't tried to re flash the card because i thought it would fix itself. But i will attempt to later on tonight. I also have a friend that his Rak miner went down. And he has a friend that has three Rak miners and they are all down. I have a freedomfi and pantherx2 and they are fine

**Comment by Jayster0966** [comment]
This happened to my goldspot last week and re-flashed latest which fixed the problem as well

**Comment by absolutebica** [comment]
> 

re-flashed latest as in the 01.04 zip off the MNTD support site?

**Comment by Jayster0966** [comment]
actually do not remember where i got the file but the file is dated 2022_02_04

**Comment by ionutnanau** [comment]
Hi, mine won't even update to the latest version after flashing with version 01.04.2022  recommanded on mntd support. It is very annoying and I don' k know why to do.

**Comment by absolutebica** [comment]
- Ok I am finally back up and running. Here are my observations. 

- January 13, 2022 I upgraded from a 32gb card to a 64gb card by flashing the 2022.01.12.1 firmware from Helium. Note, NOT Rak/MNTD, but I was under the impression they used the same file structure and files..essnetially Rak's use vanilla Helium. 
- The miner continued to perform just fine up through Feb 13, except for when the network went down sometime late Jan / early Feb. 
- It may have been the 2022.02.10.0 firmware upgrade (OTA) that threw a bunch of people, myself included, behind on the blockchain. I am not sure about this. Something happened. 
- Upon the insistence by the RakStars key moderator, Rak / MNTD miners should NOT be flashed with the original Helium firmware, and instead should use the Firmware listed on this page: https://support.getmntd.com/hc/en-us/articles/4418137820951-I-need-to-replace-the-microSD-card-on-my-hotspot (there is a note on there about the importance of this, and I don't recall seeing it in early January. I was under the impression they were essentially the same and either would work. The fact my miner continued performing fine led me to believe this is mostly true, however apparently the boot files are different
- Also, if you need Rak support to ever remotely connect to your miner, you will need the MNTD firmware, they will not be able to connect if you run other firmware. 
- So last night at 8:30pm I flashed the 2022.01.04.0 firmware listed on that site, after the new snapshot was released to the community.  I noticed immediately the firmware of 2022.02.10.0 showed up.  Those in the Rakstars discord seem to agree they saw that as well.  Shortly after, I continued to see my block height gap with the blockchain grow, the opposite of what I wanted.  Between 8:30pm and 9am the next day it had grown from 500 to 990.
- I decided to reboot this morning hoping it might kickstar a snapshot check or help clear out some junk.  Afterward the reboot, I had issues connecting via bluetooth.  Eventually I connected but had trouble running diagnostics, getting a read / characters error.  When I did get in, the inbound and outbound were grey, whereas in the past they would eventually turn green. They never did..no matter how long I waited.  I confirmed no connection via Hotspotty and the Helium Hiker Bot on discord.
- After several rounds of trying this including reboots, I decided to reflash 2022.01.04.0 again. I figured maybe when I rebooted, some important process or file was being accessed and I somehow corrupted the connection piece of the file system. I have no idea. 
- So I reflashed, and after a few minutes I noticed I had an outbound green connection. Inbound was grey (but Helium earlier was having api/status issues). Eventually I was able to see a block height and that is was growing.  
- One observation, the first time I reflashed, I saw 2022.02.10 pretty immediate, whereas this time I continued seeing the old 01.04 firmware for a while. 
- So I let it go and confirmed using HeliumGeek that my block height gap was decreasing and after a few hours, I am officially in sync and constructed a challenge. 
- I don't know why any of this happened nor why it took a 2nd reflash to get things corrected.  Maybe Balenaetcher doesn't fully wipe the card. Maybe a 2nd flash overwrote some cache or hidden bytes that still existed.  Maybe my reboot this morning happened at a sensitive time and corrupted things.  I have no idea.

**Comment by TheSDco** [comment]
Seeing the same issue.

**Comment by Jayster0966** [comment]
About 3 days ago 1 of my remote MNTD blackspot went offline. I replaced the OEM 32gb with a 64gb flashed with the file in the above post. After about 12 hours I checked and nothing has happened or improved. No inbound or outbound. Flashed card a second time and within the next 12 hours it shows as fully synced on Hotspotty. Not relayed and pings properly. (no earnings or stats though) Today I noticed that HS is slowly falling behind the blockchain again using the Heliumgeek app. At 11am -1042 blocks, 6pm -1511, then at 8pm -1607. Not sure what to do or what I can try from this point?

**Comment by gtelnet** [comment]
Not sure if this is anecdotal or not, so looking for feedback. After weeks of have all the same issues as above after flashing/replacing sd cards with any of the versions available since 01.04, including trying 02.17 on several today, I finally accomplished 100% success on four miners by adding one additional step to the flashing process. 

The one step was to use imageUSB software from PassMark to zero out the sd card before flashing the image with balenaEtcher. Since doing that, all four that I tried (two with 01.04 and two with 02.17) had zero issues pairing to BT, connecting to WiFi/Ethernet and all successfully downloaded the latest snapshot and started increasing in blocks within 5-10 minutes of powering up. 

If you're running linux, you can use a dd command to write to every sector of the sd card prior to flashing with balenaEtcher and that should accomplish the same thing.

Curious to see if that helps anyone else.

**Comment by Jayster0966** [comment]
Thanks for the tip @gtelnet. Downloaded the software and will try it and report back.

Does zero out the USB do the same thing as windows disk utility? The reason I ask is, after flashing the drive it does change the partition of the disk space so before I reflash, I usually use disk utility to bring back the SD card to its original partition then I flash again.

**Comment by gtelnet** [comment]
To my knowledge, Windows only has the ability to natively zero out a disk that has a drive letter assigned, using format with the /p:0 switch. I used that many years ago but that would require you to first delete all the partitions, create a new partition consuming the entire drive, assign it a drive letter and then run the format with /p:0. The utility I mentioned accomplishes it without requiring all those steps. I'm sure there are probably many utilities you can use to zero a drive, that's just the first one I found today. 

For my 128GB SD cards with a USB 3.0 reader and 3.0 port, the disk zeroing took just over 60 minutes. So if the Windows utility you used didn't take at least 30 minutes for a 64GB card, it wasn't doing the same thing. 

Let us know how it works out. Thanks!

**Comment by Jayster0966** [comment]
As I mentioned on Discord, this worked for me. Big thank you to @gtelnet for posting up the fix. I am now back up and running!

**Comment by gtelnet** [comment]
Jay, I just posted a blog on the topic and wanted to know if you could leave some feedback on this and let me know if there's anything missing from the directions. It's my first Helium related blog, so any comments would be appreciated. And hope your miner is still earning! https://www.gtelnetworks.com/2022/02/27/how-to-fix-your-rak-mntd-miner/

> As I mentioned on Discord, this worked for me. Big thank you to @gtelnet for posting up the fix. I am now back up and running!
> 
> Edit: Was able to sync to the BC within 4 hours and earning again!

**Comment by meowshka** [comment]
Syncing issues won't be the case after full transition of the system to the Light Hotspots, closing.

**Comment by KenDixon2** [comment]
Did you even look at the damn hotspot? 

It has been stuck on solid yellow LEDs since the 14th, and has done absolutely nothing. The dashboard won’t even load now as well. 

There are MULTIPLE users experiencing the same issue and we’ve received zero communication from this company.

I expect a response to this or we’re getting the Helium board involved.

**Comment by cokes518** [comment]
There has been multiple daily communication in our primary communication channels about this exact issue. We post status on status.helium.com as well as the #announcements channel in Discord. discord.gg/helium

we completely understand your frustration but we ask that you keep it respectful.

**Comment by KenDixon2** [comment]
There hasn’t been a Synchrobit Announcement since April 19. You have a whole discord channel full of upset customers. I would advise you to have someone, ANYONE from Synchrobit to log onto discord and begin addressing the major concerns your customers have. 

I’ll keep it respectful when your company starts treating your customers with some respect and telling us what’s going on with our orders and our flatlined hotspots.

**Comment by cokes518** [comment]
This is not syncrobit support, you'll have to reach out to them.
